common/install_devtools.R

```diff
@@ -0,0 +1,52 @@
+# This script installs version 1.13.6 of the R package devtools.
+
+# Because devtools is the tool which will install specific versions of
+# other R packages, it cannot be used it to install itself. However in
+# order to ensure there are no dependency conflicts, it also needs to
+# pin each version of its dependencies.
+
+# This script uses install.packages to install all of the dependencies
+# for devtools, however install.packages unfortunately does not
+# recursively install dependencies for the packages it
+# installs. Therefore this script installs every dependency up the
+# chain to devtools. It's not ideal, but it guarantees a stable
+# install of devtools that won't break when a new version of devtools
+# is released.
+
+
+# Treat warnings as errors, set CRAN mirror, and set parallelization:
+options(warn=2)
+options(repos=structure(c(CRAN="https://cloud.r-project.org")))
+options(Ncpus=parallel::detectCores())
+
+
+install_package_version <- function(package_name, version) {
```

Comment:
What about using the URL format `https://cran.r-project.org/package=httr&version=1.3.0` combined with an R library to find the redirect location?

What happens in the current implementation if an invalid package version is specified?

Reply:
> combined with an R library to find the redirect location

This is a chicken and the egg problem. How can we install a pinned version of that library to find the redirect location? You can see that `httr` is one of the packages being installed in this file. 

If an invalid package version is specified the package doesn't fails to install and the script will break because whatever package depends on it won't be installed.

However it would be nice if passing an invalid URL to `install.packages` would throw an error and interrupt the script rather than it just moving on. Do either you or @jaclyn-taroni maybe know a way to force it to do so?

Reply:
I assume `option(warn=2)` would do it but doesn't seem to be the case locally

Reply:
I don't necessarily _like_ this, but could do without `tryCatch` block like so:
```R
  curl_result <- system(paste0("curl --head ", package_url), intern=TRUE)
  if (grepl("404", curl_result[1])) {
    package_url <- paste0("https://cran.r-project.org/src/contrib/Archive/", package_name, "/", package_tarball)
    curl_result <- system(paste0("curl --head ", package_url), intern=TRUE)
    if (grepl("404", curl_result[1])) {
      stop(paste("Package", package_name, "version", version, "does not exist!"))
    }
  }
```